docs/docs/validating-plugin-options.md

```diff
@@ -0,0 +1,224 @@
+        message: "Hello world"
+        optionB: false, // Optional.
+      },
+    },
+  ],
+}
+```
+
+`gatsby-plugin-console-log` can ensure users have to pass a boolean to `optionA` and a string to `message`, as well as optionally pass a boolean to `optionB`, with this `pluginOptionsSchema`:
+
+```javascript:title=plugins/gatsby-plugin-console-log/gatsby-node.js
+exports.pluginOptionsSchema = ({ Joi }) => {
+  return Joi.object({
+    optionA: Joi.boolean().required().description(`Enables optionA.`),
+    message: Joi.string()
+      .required()
+      .description(`The message logged to the console.`),
+    optionB: Joi.boolean().description(`Enables optionB.`),
+  })
+}
+```
+
+If users pass options that do not match the schema, the validation will show an error when they run `gatsby develop` and prompt them to fix their configuration.
+
+## Unit testing an options schema
+
+To verify that a `pluginOptionsSchema` behaves as expected, unit test it with different configurations using the [`gatsby-plugin-utils` package](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-utils#testpluginoptionsschema).
+
+1. Add the `gatsby-plugin-utils` package to your site:
+
+   ```shell
+   npm install gatsby-plugin-utils
+   ```
+
+2. Use the `testPluginOptionsSchema` function exported from the package in your test file. For example, with [Jest](https://jestjs.io):
+
+   ```javascript:title=plugins/gatsby-plugin-console/__tests__/gatsby-node.js
+   // This is an example using Jest (https://jestjs.io/)
+   import { testPluginOptionsSchema } from "gatsby-plugin-utils"
+   import { pluginOptionsSchema } from "../gatsby-node"
+
+   describe(`pluginOptionsSchema`, () => {
+     it(`should invalidate incorrect options`, () => {
+       const options = {
+         optionA: undefined, // Should be a boolean
+         message: 123, // Should be a string
+         optionB: `not a boolean`, // Should be a boolean
+       }
+       const { isValid, errors } = testPluginOptionsSchema(
+         pluginOptionsSchema,
+         options
+       )
+
+       expect(isValid).toBe(false)
+       expect(errors).toEqual([
+         `"optionA" is required`,
+         `"message" must be a string`,
+         `"optionB" must be a boolean`,
+       ])
+     })
+
+     it(`should validate correct options`, () => {
+       const options = {
+         optionA: false,
+         message: "string",
+         optionB: true,
+       }
+       const { isValid, errors } = testPluginOptionsSchema(
+         pluginOptionsSchema,
+         options
+       )
+
+       expect(isValid).toBe(true)
+       expect(errors).toEqual([])
+     })
+   })
+   ```
+
+## Best practices for option schemas
+
+The [Joi API documentation](https://joi.dev/api/) is a great reference to use while working on a `pluginOptionsSchema` as it shows all the available types and methods.
+
+Here are some specific Joi best practices for `pluginOptionsSchema`s:
+
+- [Add descriptions](#add-descriptions)
```

Comment:
I was thinking of the list is a concise way to summarize the best practices for people to refer (back) to, rather than being a set of paragraph headers. However, I'm fine with removing it if you prefer that!

Reply:
Makes sense to me, let's leave it!